tests/validation/tests/v3_api/test_catalog_templates_ha_update.py

```diff
@@ -0,0 +1,89 @@
+import pytest
+import asyncio
+
+
+from .common import *  # NOQA
```

Comment:
This test looks like it has a few problems:

- Probably should not import that way
- Avoid the use of asyncio
- It appears this is not actually testing HA
- It appears this is not testing the change being made

Reply:
- Probably should not import that way (I know this not ideal, but just follows the code structure in the validation framework...)
- Avoid the use of asyncio (do we have any recommendation to do Asynchronous in Python?)
- the testing case is basically about to call the templateVersion API 100 times asynchronously and check if template content files are updated after the catalog repo is refreshed. because the templates content files are generated by the local cache files, and hopefully, the round-robin will get our 100 API calls route to the diffrernt rancher nodes.

Reply:
How many Rancher nodes are running? I don't see this test launching an HA cluster or communicating to the nodes. The test should not rely on 'hopefully', this is how we get flakey tests.

Reply:
The test is also testing changing the branch, which already works without this change. It needs to test updating the commit of the same branch.

Reply:
This is [validation test](https://github.com/rancher/validation) and has been moved to rancher recently, so we don't need to set the Rancher server in the code as it will be pre-configured as the environment. 
And I think you are correct about the `changing the branch will not validate it`, will remove this testing case and add it when we have a better one(updating the commit is also quite tricky to do in the validation as it do not support add git credentails)

Reply:
From looking at the validation test scripts it appears that it launches a single node Rancher server. 

The test could checkout a specific commit of the cache on the HA nodes and verify that the server updates it on catalog change, rather than trying to modify a git repository.